src/ripple/app/paths/PathRequest.cpp

```diff
@@ -585,10 +604,20 @@ PathRequest::findPaths (std::shared_ptr<RippleLineCache> const& cache, int const
         }
     }
 
+    // Charge cost is based on source currencies used.
+    double const min_cost = 50;
+    double const max_cost = 400;
+    Resource::Charge c {static_cast<int>(
+        std::min(max_cost, std::max(
+            min_cost, std::pow(sourceCurrencies.size(), 2)))),
+                "path update"};
+    consumer_.charge(c);
+
```

Comment:
@seelabs An assert nor check is necessary. I doubt we will ever go beyond 400, which is already high. If someone used a greater number, say 1000 (absurdly high and incorrect), the code would disconnect the connection.